docs/0002-choose-a-pagination-strategy-and-implementation.md

```diff
@@ -0,0 +1,101 @@
+
+**Consistent Ordering**: even when data is added/removed results are stable
+
+**Can page forward and backwards**: tokens can be used to page before and after search results
+
+**Implementation logic hidden from users**: while there may be an interpretation of a continuation token, it is not
+guaranteed to be consistent or long-lived, which frees up back-end optimizations and changes without breaking the API
+
+#### Disadvantages
+**Implementation more complex**: needs to manage sort ordering to ensure that the continuation token equality check is
+consistent even if data is added, removed, or sorted by additional fields
+
+**Tokens become invalid**: continuation tokens can become invalid, users should not rely on them for long periods of 
+time
+
+## Decision
+
+We will use **continuation** tokens for managing pagination. The advantage of keeping the implementation flexible/hidden
+from end-users so that it can be optimized or changed later without breaking functionality for existing users outweighs
+disadvantages around increased complexity for implementation. Additionally, since `franlin` APIs are expected to hold 
+large amounts of data (tens of millions of rows) it is important that the pagination approach scales accordingly.
+Lastly, since eventually `franklin` may move beyond relying solely on `postgresl` for managing state it may actually be
+better that we do not use `offset/limit` which relies heavily on `SQL`. 
+
+## Consequences
+
+An upcoming release for the [STAC specification](https://github.com/radiantearth/stac-spec/blob/dev/CHANGELOG.md#changed) 
+changes how pagination will be communicated in the API and simplifies implementation. In order to implement this we will 
+need update API responses for `/search` and `/collections/<id>/items/` to include a `next` link in its responses for
+paginating forward. Initial implementation should `base64` encode the primary key and for the last result in the current
+page.
+
+One thing that is unclear is how we should handle primary keys/identifiers when implementing `continuation` pagination.
+This approach to pagination relies on sorting by an identifier that signals the last value in the current page. If that 
+value is a primary key then then including the PK in the `sort by` for `SQL` queries with a limit gives the next set of 
+results. For a `SERIAL` column in `postgresql` this is performant with an index. However, we use client supplied 
+identifiers right now for IDs since they are likely meaningful to the client and don't create our own PK. In 
+implementing a `continuation` token based approach we should keep the user supplied identifier, but also add a column 
+for a real PK that is a `SERIAL` column.
+
+The implementation should follow the approach laid out [here](https://phauer.com/2018/web-api-pagination-timestamp-id-continuation-token/) 
+every query ends up being something like:
+
+```SQL
+SELECT * FROM table
+WHERE 
+  id > <continuation-id> 
```

Comment:
I had trouble reading this the first time because of the mix of `>` with angle brackets for interpolation -- could we use braces for interpolation instead?
```suggestion
  id > {continuation-id} 
```